# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Lane: Refresh / Content Opportunity Scoring**

I'm choosing this lane because it maps directly onto a real, recurring decision: which content pages should a reviewer look at first when there isn't enough time to review everything. The starter dataset already contains the signals needed for this (impressions, sessions, position, freshness, engagement), and the starter pipeline gives me a working baseline model to compare against, so I can build on something concrete rather than starting from zero.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Question:** Which pages should be reviewed first for refresh, expansion, protection, pruning, or monitoring?

**Unit of analysis:** One content page (content_id), for a given client.

**Decision:** A content reviewer with limited time must choose which pages to spend their attention on first.

**Action:** Based on the recommendation, the reviewer refreshes, rewrites, expands, protects, or simply monitors a page.

**Cost of a wrong call:**
- False positive (flagged but page is actually fine): wastes reviewer time — a moderate cost, since review capacity is limited.
- False negative (a genuinely declining, high-demand page is missed): traffic loss goes unnoticed and unaddressed — potentially a larger cost.
- Because both errors matter, the ranking needs to be both precise at the top and not blind to real high-value opportunities.

**Why data/ML helps:** There are hundreds of thousands of content pages — far too many for manual review. A data-driven ranked score lets a human reviewer focus limited attention where it's statistically most likely to matter, instead of guessing or reviewing pages in arbitrary order.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
!git clone https://github.com/mahmii12/FlyRank-Assignment-.git
%cd FlyRank-Assignment-
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total rows:", len(df))
print("Rows with impressions_90d > 0:", (df['impressions_90d'] > 0).sum())
print("Rows currently declining (trend_direction == 'down'):",
      (df['trend_direction'] == 'down').sum())
print("Rows that are both visible AND declining:",
      ((df['impressions_90d'] > 0) & (df['trend_direction'] == 'down')).sum())

Cloning into 'FlyRank-Assignment-'...
remote: Enumerating objects: 108, done.
remote: Counting objects: 100% (108/108), done.
remote: Compressing objects: 100% (79/79), done.
remote: Total 108 (delta 26), reused 79 (delta 13), pack-reused 0 (from 0)
Receiving objects: 100% (108/108), 1.85 MiB | 6.06 MiB/s, done.
Resolving deltas: 100% (26/26), done.
/content/FlyRank-Assignment-/FlyRank-Assignment-/FlyRank-Assignment-/FlyRank-Assignment-/FlyRank-Assignment-/FlyRank-Assignment-
Total rows: 30000
Rows with impressions_90d > 0: 30000
Rows currently declining (trend_direction == 'down'): 16262
Rows that are both visible AND declining: 16262


Out of **30,000** pages in the starter dataset, all **30,000** have real search visibility (impressions_90d > 0), and **16,262** of them are currently in a declining trend — more than half the dataset. This confirms there is a large, meaningful pool of realistic review candidates for this lane, not just a handful of edge cases.

This is also backed by the starter pipeline's own model results: a random forest model reached Precision@50 = 0.740, versus 0.240 for a simple fixed rule — meaning a learned ranking correctly caught about 37 of its top 50 recommended pages, versus about 12 for the baseline rule. That's strong evidence a learned approach is worth the next 7 weeks of effort in this lane.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**I can claim:** which pages show statistical evidence of decline, demand, or opportunity, based on observed historical signals (impressions, sessions, position, engagement).

**I cannot claim:** that refreshing a flagged page will *cause* a recovery. Proving causation would require a controlled experiment (e.g. A/B testing refreshed vs. non-refreshed pages), which this dataset does not provide.

**I also cannot claim:** to have reverse-engineered any search engine's ranking algorithm. My model only learns patterns from FlyRank's own observed data, not Google's internal logic.

**I will use language like** "this suggests," "we observed," or "this is associated with" — not "this proves" or "this causes."

## Self-check

Before you submit, confirm each line honestly:

- [X] Every section above is filled — markdown thinking AND the code that backs it
- [X] The notebook runs top to bottom with no errors (Runtime → Run all)
- [X] No client names, URLs, or private queries anywhere
- [X] My claims use careful words: observed, measured, directional, decision-support
- [X] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.